utils/chunked_vector.hh

```diff
@@ -325,8 +327,32 @@ size_t chunked_vector<T, max_contiguous_allocation>::external_memory_usage() con
 template <typename T, size_t max_contiguous_allocation>
 chunked_vector<T, max_contiguous_allocation>::chunked_vector(const chunked_vector& x)
         : chunked_vector() {
-    reserve(x.size());
-    std::copy(x.begin(), x.end(), std::back_inserter(*this));
+    auto size = x.size();
+    if (!size) {
+        return;
+    }
+    reserve(size);
+    auto copy_chunk = [this, &x] (size_t i, size_t n) {
+        const T* src = x._chunks[i].get();
+        T* dst = _chunks[i].get();
+        if constexpr (std::is_trivially_copyable_v<T>) {
+            std::memcpy(dst, src, n * sizeof(T));
+        } else {
+            // dst isn't initialized so can't do array copy
+            for (auto end = src + n; src != end; ++dst, ++src) {
+                new (dst) T(*src);
+            }
+        }
```

Comment:
This is std::uninitialized_copy() (which includes the optimization for trivially copyable types).

Reply:
done in v2